cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -233,6 +235,34 @@ def _set_default_instance_role(self):
             ],
         }
 
+        if build.components:
+            for custom_component in build.components:
+                # Check custom component is script, and the url is S3 url
+                if custom_component.type == "script" and (
+                    custom_component.value.startswith("s3") or check_https_url_type(custom_component.value)
+                ):
+                    bucket_info = get_bucket_object_name(custom_component.value)
+                    bucket_name = bucket_info.get("bucket_name")
+                    object_name = bucket_info.get("object_name")
+                    try:
+                        # If the object is public, s3 get object policy is not required
+                        response = AWSApi.instance().s3.get_object_acl(bucket_name=bucket_name, object_name=object_name)
```

Comment:
Object can be public even if the object ACL is not public

Reply:
done, always create policy for s3 url